### ISSUE TITLE
Cannot create new visualizations to Chartbrew dashboards using UI.

### ISSUE THREAD

**Issue by bachdgvn** [chartbrew/strapi-plugin-chartbrew#11]
Cannot create new visualizations to Chartbrew dashboards using UI.

![image](https://user-images.githubusercontent.com/30729887/200102375-953dfb06-dfd4-4047-910e-f60878786517.png)

This is my strapi server log.
```
[2022-11-05 12:04:25.203] error: 400 - "Missing required arguments to create the template"
StatusCodeError: 400 - "Missing required arguments to create the template"
    at new StatusCodeError (F:\Learn\my-project\node_modules\request-promise-core\lib\errors
.js:32:15)
    at Request.plumbing.callback (F:\Learn\my-project\node_modules\request-promise-core\lib\
plumbing.js:104:33)
    at Request.RP$callback [as _callback] (F:\Learn\my-project\node_modules\request-promise-
core\lib\plumbing.js:46:31)
    at Request.self.callback (F:\Learn\my-project\node_modules\request\request.js:185:22)
    at Request.emit (node:events:390:28)
    at Request.emit (node:domain:475:12)
    at Request.<anonymous> (F:\Learn\my-project\node_modules\request\request.js:1154:10)
    at Request.emit (node:events:390:28)
    at Request.emit (node:domain:475:12)
    at IncomingMessage.<anonymous> (F:\Learn\my-project\node_modules\request\request.js:1076
:12)
    at Object.onceWrapper (node:events:509:28)
    at IncomingMessage.emit (node:events:402:35)
    at IncomingMessage.emit (node:domain:475:12)
    at endReadableNT (node:internal/streams/readable:1343:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
From previous event:
    at Request.plumbing.init (F:\Learn\my-project\node_modules\request-promise-core\lib\plum
bing.js:36:28)
    at Request.RP$initInterceptor [as init] (F:\Learn\my-project\node_modules\request-promis
e-core\configure\request2.js:41:27)
    at new Request (F:\Learn\my-project\node_modules\request\request.js:127:8)
    at request (F:\Learn\my-project\node_modules\request\index.js:53:10)
    at Object.generate (F:\Learn\my-project\node_modules\@chartbrew\plugin-strapi\server\ser
vices\chartbrew.js:111:11)
    at async Object.generateTemplate (F:\Learn\my-project\node_modules\@chartbrew\plugin-str
api\server\controllers\chartbrew.js:64:23)
    at async returnBodyMiddleware (F:\Learn\my-project\node_modules\@strapi\strapi\lib\servi
ces\server\compose-endpoint.js:52:18)
    at async policiesMiddleware (F:\Learn\my-project\node_modules\@strapi\strapi\lib\services\server\policy.js:24:5)
    at async F:\Learn\my-project\src\middlewares\creditCollector.js:12:5
    at async F:\Learn\my-project\node_modules\@strapi\strapi\lib\middlewares\body.js:58:9
    at async F:\Learn\my-project\node_modules\@strapi\strapi\lib\middlewares\logger.js:25:5
    at async F:\Learn\my-project\node_modules\@strapi\strapi\lib\middlewares\powered-by.js:16:5
    at async cors (F:\Learn\my-project\node_modules\@koa\cors\index.js:107:16)
    at async F:\Learn\my-project\node_modules\@strapi\strapi\lib\middlewares\errors.js:13:7
    at async session (F:\Learn\my-project\node_modules\koa-session\index.js:41:7)
    at async F:\Learn\my-project\node_modules\@strapi\strapi\lib\services\metrics\middleware.js:29:5

```

**Comment by razvanilin** [comment]
Hey @bachdgvn This usually means that Chartbrew cannot reach your Strapi instance.

For example, if you are running Strapi locally on your machine and Chartbrew is on another machine, it cannot create the dashboard for you. Can you give me more details about your current setup?

**Comment by razvanilin** [comment]
Hey @bachdgvn I was finally able to replicate this issue and it was due to an environmental not being set up in all Strapi production instances. The variable holds the Strapi Backend URL which is needed by Chartbrew to know where to get the data from.

I just changed released v0.4.0 which adds a new field in the plugin settings in the admin dashboard where you can set your backend URL manually.

![CleanShot 2022-11-23 at 17 02 49@2x](https://user-images.githubusercontent.com/5342321/203521095-36228c81-428a-41f9-a27a-0ddab85762fa.jpg)

I'll close this issue, but feel free to re-open if you still encounter the issue
